Upgrade dependencies

alse opened the pull request:
#324

- Upgrades all dependencies except those related to Apollo and monaco-languageclient + vscode-jsonrpc
- Upgrading monaco-languageclient and switching to vscode-ws-jsonrpc is a larger task we can break out
- Upgrading to Apollo 3 is also a larger task that will require some refactoring and using hooks instead of the projectMutator
- React was only upgraded to 17. React 18 caused a linting regression:
    1. Open a new project
    2. Deploy a HelloWorld contract
    3. Click on the Transaction template
    4. The HelloWorld contract is not found on 0x01 until a page reload


______

For contributor use:

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/onflow/flow-playground/blob/master/CONTRIBUTING.md))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [standards mentioned here](https://github.com/onflow/flow-playground/blob/master/CONTRIBUTING.md).
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

alse committed:
Upgrade dependencies

MrDSGC commented on src/components/AccountAvatars.tsx:
```diff
@@ -29,7 +33,7 @@ export const AccountAvatar: React.FC<{
 };
 
-export const AvatarList: React.FC = ({ children }) => {
+export const AvatarList = ({ children }: { children: React.ReactNode }) => {
```
Does it make sense to make a `ChildProps` ~component~ thats type `{children}` ? i see it being used else where

alse replied:
Do you mean a ChildProps component, or a ChildProps type?

MrDSGC replied:
type sorry

MrDSGC commented on package.json:
```diff
@@ -2,7 +2,7 @@
   "private": true,
   "scripts": {
-    "start": "webpack-dev-server -d --config webpack.config.js --inline ",
+    "start": "webpack-dev-server -d source-map --mode=development --config webpack.config.js",
```
do you mind explaining how development mode works since i didnt see the mode in the webpack config

alse replied:
The latest webpack throws an error without an explicit mode. The mode can be set here, or we can add a default mode to webpack.config.js. Let me know if you prefer that

MrDSGC replied:
yeah lets add a default mode for readability

MrDSGC commented on src/components/AccountSigners.tsx:
```diff
@@ -5,7 +5,7 @@ import React from 'react';
 import { Badge, Flex, Text, useThemeUI } from 'theme-ui';
 
-export const Outline: React.FC = ({ children }) => {
+export const Outline = ({ children }: { children: React.ReactNode }) => {
```
same childProps comment

MrDSGC commented on src/components/EditorPanel/components.tsx:
```diff
@@ -62,9 +62,12 @@ export const PanelContainer = styled(motion.div)`
 `;
 
-export const PanelFooter: React.FC<{ dismiss: () => void }> = ({
+export const PanelFooter = ({
```
same childProps comment, this could be an extentension of the childProps type?

MrDSGC commented on src/components/EditorPanel/components.tsx:
```diff
@@ -147,7 +150,7 @@ export const PanelFooter: React.FC<{ dismiss: () => void }> = ({
 };
 
-export const EditorHeader: React.FC = ({ children }) => {
+export const EditorHeader = ({ children }: { children: React.ReactNode }) => {
```
same childProps comment

MrDSGC commented on src/components/EditorPanel/index.tsx:
```diff
@@ -2,15 +2,15 @@ import React from 'react';
 } from './components';
 
-const EditorPanel: React.FC = ({ children }) => {
+const EditorPanel = ({ children }: { children: React.ReactNode }) => {
```
same childProps comment

MrDSGC commented on src/components/EditorPanel/components.tsx:
```diff
@@ -164,7 +167,11 @@ export const EditorHeader: React.FC = ({ children }) => {
 
-export const EditorPanelMenu: React.FC = ({ children }) => {
+export const EditorPanelMenu = ({
+  children,
```
same childProps comment

MrDSGC commented on src/containers/Editor/components.tsx:
```diff
@@ -46,7 +46,7 @@ export interface WithShowProps {
 }
 
-const Header: React.FC = ({ children }) => {
+const Header = ({ children }: { children: React.ReactNode }) => {
```
same childProps comment

MrDSGC commented on src/containers/Editor/components.tsx:
```diff
@@ -105,13 +105,19 @@ const ShareButton: React.FC<{ url: string }> = ({ url }) => {
 };
 
-const ShareSaveButton: React.FC<{
+const ShareSaveButton = ({
```
Move this to its own proptype?

MrDSGC commented on webpack.config.js:
```diff
@@ -64,11 +64,13 @@ module.exports = {
     ],
   },
   devServer: {
```
does this map over to `mode=development` in the webpack build command?

alse replied:
Yes, these are the dev server options

MrDSGC reviewed (approved):
small aesthetic comments and a few clarifying questions. then lgtm. I woud like to create type vars to plug in instead of directly passing types into functions because i want to have a centralized types file for the redesign. would like to hear your thoughts on this during our sync as well

alse commented:
@MrDSGC centralized types sound good for shared/exported types. I usually keep component types local. Thanks for reviewing!

alse committed:
PR feedback
- create shared children prop type
- set default webpack env in webpack.config.js
